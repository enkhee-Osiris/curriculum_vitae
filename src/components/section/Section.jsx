import React from 'react';
import { node } from 'prop-types';

const Section = ({ children }) => (
  <section>
    {children}
  </section>
);

Section.propTypes = {
  children: node.isRequired,
};

export { Section as default };
