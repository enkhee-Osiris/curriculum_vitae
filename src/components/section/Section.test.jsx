import React from 'react';
import { shallow } from 'enzyme';
import Section from './Section';

describe('<Section />', () => {
  it('should be defined', () => {
    expect(Section).toBeDefined();
  });

  it('renders correctly', () => {
    const children = <div>test</div>;
    const section = shallow(<Section> {children} </Section>);

    expect(section).toMatchSnapshot();
  });

  it('renders children', () => {
    const children = <div>test</div>;
    const section = shallow(<Section> {children} </Section>);

    expect(section.contains(children)).toEqual(true);
  });
});
