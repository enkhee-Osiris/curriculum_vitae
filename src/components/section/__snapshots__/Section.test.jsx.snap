// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section>
     
    <div>
        test
    </div>
     
</Section>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        <div>
          test
</div>,
        " ",
      ],
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      " ",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <div>
            test
</div>,
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
        " ",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
